src/main/java/io/kamax/matrix/client/MatrixHttpRoom.java

```diff
@@ -63,6 +63,7 @@ public MatrixHttpRoom(MatrixClientContext context, String roomId) {
     protected URIBuilder getClientPathBuilder(String action) {
         URIBuilder builder = super.getClientPathBuilder(action);
         builder.setPath(builder.getPath().replace("{roomId}", roomId));
+        context.getUser().ifPresent(user -> builder.setPath(builder.getPath().replace("{userId}", user.getId())));
```

Comment:
Good question. It has nothing to do with the PR itself, but I needed it to create the test cases, which in turn I needed to verify the changes in this PR.

Reply:
Should I move this to another PR?

Reply:
It feels like this should be into the Abstract class instead of this specific one, where it would be totally valid in case any concrete implementation needs to map `{userId}`. Doing a specific PR for this seems overkilling it, but the code is definitely at the wrong place.

Putting it at the right place would make it acceptable for here.

Reply:
Currently the user id gets replaced in `AMatrixHttpClient`, when the user is a virtual user. I think, that's not the case here. Do you remember, why it only gets replaced for virtual users? Can I remove that if-clause from the replacement? See line 359 in `AMatrixHttpClient`.